facade/src/main/scala/reactST/reactTable/TableMaker.scala

```diff
@@ -179,8 +193,101 @@ trait TableMaker[
     component(options)
   }
 
+  /**
+   * Create a table element based on the configured plugins.
+   * The plugins MUST include withBlockLayout.
+   *
+   * @param options The table options to use for the table.
+   * @param data The table data.
+   * @param rowIdFn Function to extract the row id from a data element. See note.
+   * @param rowHeight The height of each row.
+   * @param bodyHeight The height of the table body (virtualized portion).
+   * @param headerCellFn See note. A function to use for creating the header elements. Simple ones are provided by the TableMaker object.
+   * @param tableClass An optional CSS class to apply to the table element
+   *
+   * Notes:
+   * Virtualized tables are not built from html table elements such as
+   * <table>, <tr>, etc. The virtualization requires they be made from
+   * <div> elements and styled as a table. This method adds a class to
+   * each div that corresponds to it's role, such as "table", "thead",
+   * "tr", etc.
+   * This means that headerCellFn must also return a <div> instead
+   * of a <td>, and should probably also have a class of "td". The basic
+   * header functions in TableMaker take a boolean flag to handle this.
+   * The rowIdfn is applied to both the the table options and the react-window
+   * virtualization. So, you don't need to apply it to the options beforehand.
+   */
+  def makeVirtualizedTable(
+    options:           TableOptsD,
+    data:              js.Array[D],
+    rowIdFn:           D => String,
+    rowHeight:         Double,
```

Comment:
I wonder if this can be `D => Double` or something like that for tables that have different heights per row. Or maybe it is supported some other way?

Reply:
You can use a function `(index: number) => number` if you use a VariableSizeList from ReactWindow instead of a FixedSizeList which is what we use here.

The two options I see to support this are:

1. always use VariableSizeList and just have a function that ignores the index for tables where the height is always the same.
2. have a separate method for creating tables with variably sized rows.
I don't know if there are performance issues with the variable sizes, but I wouldn't think so. It seems like it would just be a function call per row. If we think this will be a common scenario, we should probably go with option 1 for simplicity of use.